modules/extensions-support/src/main/java/org/mule/module/extension/internal/util/IntrospectionUtils.java

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.util;
+
+import static org.apache.commons.lang.StringUtils.EMPTY;
+import static org.mule.module.extension.internal.introspection.MuleExtensionAnnotationParser.getMemberName;
+import static org.mule.util.Preconditions.checkArgument;
+import static org.reflections.ReflectionUtils.getAllFields;
+import static org.reflections.ReflectionUtils.getAllMethods;
+import static org.reflections.ReflectionUtils.withAnnotation;
+import static org.reflections.ReflectionUtils.withModifier;
+import static org.reflections.ReflectionUtils.withName;
+import static org.reflections.ReflectionUtils.withParameters;
+import static org.reflections.ReflectionUtils.withTypeAssignableTo;
+import org.mule.api.NestedProcessor;
+import org.mule.extension.annotations.param.Ignore;
+import org.mule.extension.annotations.param.Optional;
+import org.mule.extension.introspection.DataType;
+import org.mule.extension.introspection.Operation;
+import org.mule.extension.introspection.Parameter;
+import org.mule.util.ArrayUtils;
+import org.mule.util.ClassUtils;
+import org.mule.util.CollectionUtils;
+
+import java.lang.reflect.AccessibleObject;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.Collection;
+import java.util.List;
+
+import org.apache.commons.lang.StringUtils;
+import org.springframework.core.ResolvableType;
+
+/**
+ * Set of utility operations to get insights about objects and their operations
+ *
+ * @since 3.7.0
+ */
+public class IntrospectionUtils
+{
+
+    /**
+     * Returns a {@link DataType} representing the
+     * given clazz
+     */
+    public static DataType getClassDataType(Class<?> clazz)
```

Comment:
This class seems to manage different aspects of "introspection". Many methods are utilities for DataType, others for class,etc. Can you split it?